internal/cmd/options.go

```diff
@@ -48,3 +49,22 @@ func NewLeaderElectionOptions() (LeaderElectionOptions, error) {
 	}
 	return leaderOpts, nil
 }
+
+func ParseEnvAgentReplicaCount() (int32, error) {
+	replicas, err := parseEnvInt32("FLEET_AGENT_REPLICA_COUNT")
+	if err != nil {
+		return 0, err
+	}
+	return replicas, nil
+}
+
+func parseEnvInt32(envVar string) (int32, error) {
+	if d := os.Getenv(envVar); d != "" {
```

Comment:
nit: this could make use of [os.Lookupenv](https://pkg.go.dev/os#LookupEnv) and check its 2nd return parameter which indicates whether the variable is set.